src/moments.jl

```diff
@@ -63,10 +63,18 @@ function var!(R::AbstractArray, A::RealArray, wv::WeightVec, dim::Int; mean=noth
 end
 
 Base.varm(A::RealArray, wv::WeightVec, M::RealArray, dim::Int) =
-    Base.varm!(Array(Float64, Base.reduced_dims(A, dim)), A, wv, M, dim)
+    @static if VERSION < v"0.5.1-pre+19"
+        Base.varm!(similar(A, Float64, Base.reduced_dims(size(A), dim)), A, wv, M, dim)
+    else
+        Base.varm!(similar(A, Float64, Base.reduced_indices(indices(A), dim)), A, wv, M, dim)
```

Comment:
It depends on how often these are used. It is typically much simpler to add a few conditions like these than adding functionality to `Compat` because it has to be more general, tested and documented. If the functionality gets added to `Compat`, we can update the code here but the priority right now is to get the stats package to work on master.